source/common/tracing/zipkin/zipkin_tracer_impl.cc

```diff
@@ -81,30 +79,17 @@ Tracing::SpanPtr Driver::startSpan(const Tracing::Config& config, Http::HeaderMa
                                    const std::string&, SystemTime start_time) {
   Tracer& tracer = *tls_->getTyped<TlsTracer>().tracer_;
   SpanPtr new_zipkin_span;
+  bool sampled(true);
+
+  if (request_headers.XB3Sampled()) {
+    // Checking if sampled flag has been specified. Also checking for 'true' value, as some old
+    // zipkin tracers may still use that value, although should be 0 or 1.
+    sampled =
+        request_headers.XB3Sampled()->value().getString() == ZipkinCoreConstants::get().SAMPLED ||
+        request_headers.XB3Sampled()->value().getString() == "true";
+  }
 
-  if (request_headers.OtSpanContext()) {
-    // Get the open-tracing span context.
```

Comment:
I'm not sure how this code removal relates to X-B3-Sampled header.  Can you clue me in, or possibly extend the PR description?

Reply:
This is because there is no way to encode the B3 sampling flag in the `OtSpanContext` in a backward compatible way. So instead, by no longer using the `OtSpanContext` header with the zipkin tracer, it will just fallback to the information in the B3 headers.

Reply:
Ah sorry, I now see you called that out earlier on in the review.  I'm happy with Matt and Adrian's sign off here as long as the addition and the deprecation both make the new style release notes Matt asked for (https://github.com/envoyproxy/data-plane-api/blob/master/docs/root/intro/version_history.rst)

Reply:
Oh also if you could link to release notes and docs updates in the top level template comment it'd be super helpful for merging :-)

Reply:
Version history has been updated. Will add link to doc/rel note PR.